R/parser.R

```diff
@@ -459,7 +459,9 @@ token <- function(p) {
     p %then%
     space() %using%
     function(x) {
-      unlist(c(x))[2]
+      x <- x[-1]
```

Comment:
The general idea of the change was intentional, and I'm curious which way you'd like for it to be fixed.  With my proposed changes, nearly every result by default will be a list.  That enables a consistent interface to results and protects against accidental loss of attributes.  The updated Unlist function preserves that notion.

It is feasible, though a bit more complex, to promote results out of being lists if feasible, but it requires more checks and then the user would have to know more about the object they were about to retrieve (e.g. "Am I about to receive a list, a character string, ...?").

My preference would be to fix it by updating the tests to assume that `result` is always a list-- maybe with 0, 1, or more than 1 item.  If that works for you, I can make the change (which will largely be updating the tests to align with that decision).  If that's not your preference, I can make the updates so that a variety of variable types may be within `result`.

(Neither of these choices would take away the option for the user to write their own functions-- typically with `%using%`-- to update `result` to whatever they want.)